docs/dev_cycle.part-04.rst

```diff
@@ -0,0 +1,315 @@
+        donations_count = donator.call().donations_count()
+        donations_total = donator.call().donations_total()
+        default_usd_rate = donator.call().default_usd_rate()
+
+        assert donations_total == 1150
+        assert donations_count == 2
+        assert default_usd_rate == 380
+
+
+You added another test, ``test_donations``. The second test is similar to the first one:
+
+**[1] Get the chain**: The test function accepts the ``chain`` argument, the auto-generated Python object that
+corresponds to a ``tester`` chain. Reminder: the ``tester`` chain is ephimeral, in memory, and reset
+on each test function.
+
+**[2] Get the contract**: With the magic function ``get_or_deploy_contract`` Populus compiles the `Donator` contract,
+deploys it to the ``chain``, creates a Web3 ``contract`` object, and returns it to the function as a Python
+object with Python methods. This object is stored in the ``donator`` variable.
+
+**[3] The "transact" function**:
+
+.. code-block:: python
+
+    donator.transact({'value':500}).donate(370)
+
+Reminder: we have two options to interact with a contract on the blockchain, *transactions* and *calls*.
+With Populus, you initiate a transaction with ``transact``, and a call with ``call``:
+
+* *Transactions*: Send a transaction, run the contract code, transfer funds, and *change* the state of the contract and it's balance. This change will be permenant, and synced to the entire blockchain.
+
+* *Call*: Behaves exactly as a transaction, but once done, everything is revert and no state is changed. A call is kinda "dry-run"
+
+
+**[4] Test transactions**: The test commits two transactions, and send funds in both. In the first the ``value`` of the funds is 500,
+and in the second the ``value`` is 650.
+The ``value`` is provided as a ``transact`` argument, in a dictionary, where you can add more kwargs of an Ethereum
+transaction.
+
+
+.. note::
+
+    Since these are *transactions*, they will change state, and in the case of the ``tester`` chain this state will persist
+    until the test function quits.
+
+
+**[5] Providing arguments**: The donate function in the contract accepts one argument
+
+.. code-block:: solidity
+
+    function donate(uint usd_rate) public payable money_sent {...}
+
+This argument is provided in the test as *Python* donate function:
+
+.. code-block:: python
+
+    donator.transact({'value':650}).donate(380).
+
+Populus gives you a *Python* interface to a bytecode contract. Nice, no?
+
+**[6] Asserts**: We expect the ``donations_total`` to be ``500 + 650 = 1150``, the ``donations_count`` is 2,
+and the ``default_usd_rate`` to match the last update, 380. The test gets the varaibles with ``call``, but in the case
+of a "getter" it doesn't matter, since a "getter" function will not change the contract's state wether it's called with ``transact`` or ``call``.
+
+Run the test:
+
+.. code-block:: bash
+
+    $ py.test --disable-pytest-warnings
```

Comment:
Why the `--disable-pytest-warnings`?

Reply:
unrelated warnings which may be confusing. will add a note